src/rgw/setup/rgw.py

```diff
@@ -494,48 +494,46 @@ def _update_hax_endpoint_and_create_admin(conf: MappedConf):
+        # 4. If user creation is successful, update lock value to 'user_created'.
 
         rgw_lock = False
-        rgw_lock_key = f'component>{COMPONENT_NAME}>volatile>{COMPONENT_NAME}_lock'
         rgw_consul_idx = f'{COMPONENT_NAME}_consul_idx'
         # Get consul url from cortx config.
         consul_url = Rgw._get_consul_url(conf)
         # Check for rgw_lock in consul kv store.
         Log.info('Checking for rgw lock in consul kv store.')
-        Conf.load(rgw_consul_idx, consul_url)
-
+        Rgw._load_rgw_config(rgw_consul_idx, consul_url)
         # if in case try-catch block code executed at the same time on all the nodes,
         # then all nodes will try to update rgw lock-key in consul, after updating key
         # it will wait for sometime(time.sleep(3)) and in next iteration all nodes will
         # get lock value as node-id of node who has updated the lock key at last.
         # and then only that node will perform the user creation operation.
         while True:
             try:
-                rgw_lock_val = Conf.get(rgw_consul_idx, rgw_lock_key)
-                Log.info(f'rgw_lock value - {rgw_lock_val}')
-                # TODO: Explore consul lock - https://www.consul.io/commands/lock
+                rgw_lock_val = Conf.get(rgw_consul_idx, CONSUL_LOCK_KEY)
+                Log.info(f'{CONSUL_LOCK_KEY} value - {rgw_lock_val}')
                 if rgw_lock_val is None:
-                    Log.info(f'Setting confstore value for key :{rgw_lock_key}'
-                             f' and value as :{Rgw._machine_id}')
-                    Rgw._load_rgw_config(rgw_consul_idx, consul_url)
-                    Conf.set(rgw_consul_idx, rgw_lock_key, Rgw._machine_id)
-                    Conf.save(rgw_consul_idx)
-                    Log.info(f'Updated confstore with latest value {Rgw._machine_id}')
-                    time.sleep(3)
+                    Log.info(f'Setting consul kv store value for key :{CONSUL_LOCK_KEY}'
+                            f' and value as :{Rgw._machine_id}')
+                    Rgw._set_consul_kv(rgw_consul_idx, CONSUL_LOCK_KEY, Rgw._machine_id)
                     continue
                 elif rgw_lock_val == Rgw._machine_id:
-                    Log.info('Found lock acquired successfully hence processing'
-                             ' with RGW admin user creation.')
+                    Log.info('Found lock acquired successfully, hence proceeding'
+                            f' with RGW admin user creation on node {rgw_lock_val}.')
                     rgw_lock = True
                     break
                 elif rgw_lock_val != Rgw._machine_id:
-                    Log.info(
-                        'Skipping rgw user creation, as rgw lock is already'
-                        f' acquired by other machine with id {rgw_lock_val}')
-                    rgw_lock = False
-                    break
+                    if rgw_lock_val == 'user_created':
```

Comment:
can we add this string 'user_created' as constants ? 
tomorrow we may need to change it through const.py